pybaseball/teamid_lookup.py

```diff
@@ -57,56 +93,110 @@ def _generate_teams() -> pd.DataFrame:
-                                              suffixes=['', '.p'])
-
-    # Drop lahman data down to just what we need
-    teams_franchises = teams_franchises[
-        ['yearID', 'lgID', 'teamID', 'franchID', 'divID', 'name', 'park', 'teamIDBR', 'teamIDlahman45', 'teamIDretro',
-         'franchName', 'city', 'state']
-    ]
-
-    # Try to guess the name Fangraphs would use
-    teams_franchises['possibleName'] = teams_franchises.apply(lambda row: _estimate_name(row, 'name'), axis=1)
-    teams_franchises['possibleFranchName'] = teams_franchises.apply(lambda row: _estimate_name(row, 'franchName'),
-                                                                    axis=1)
-
-    # Join up the data by team name, and look for what is still without a match
-    outer_joined = teams_franchises.merge(team_data, how='outer', left_on=['yearID', 'possibleName'],
-                                          right_on=['Season', 'Team'])
-    unjoined_teams_franchises = outer_joined.query('Season.isnull()').drop(team_data.columns, axis=1)
-    unjoined_team_data = outer_joined.query('yearID.isnull()').drop(teams_franchises.columns, axis=1)
-
-    # Take all the unmatched data and try to join off franchise name, instead of team name
-    inner_joined = teams_franchises.merge(team_data, how='inner', left_on=['yearID', 'possibleName'],
-                                          right_on=['Season', 'Team'])
-    franch_inner_joined = unjoined_teams_franchises.merge(unjoined_team_data, how='inner',
-                                                          left_on=['yearID', 'possibleFranchName'],
-                                                          right_on=['Season', 'Team'])
+    fg_team_data = fangraphs.fg_team_batting_data(start_season, end_season, "ALL", stat_columns=['AB'])
+
+    fg_columns = list(fg_team_data.columns.values)
+
+    unjoined_team_data = fg_team_data.copy(deep=True)
+
+    unjoined_lahman_teams = lahman_teams.copy(deep=True)
+
+    unjoined_lahman_teams['manual_teamid'] = unjoined_lahman_teams.apply(
+        lambda row: _manual_matches.get(row.franchID, -1),
+        axis=1
+    )
+
+    lahman_columns += ['manual_teamid']
+
+    unjoined_lahman_teams['initials'] = unjoined_lahman_teams.apply(
+        lambda row: re.sub(r'[^A-Z]', '', row['name']),
+        axis=1
+    )
+
+    lahman_columns += ['initials']
+
+    unjoined_lahman_teams['city_start'] = unjoined_lahman_teams.apply(
+        lambda row: row['name'][:3].upper(),
+        axis=1
+    )
+
+    lahman_columns += ['city_start']
+
+    unjoined_lahman_teams['name_start'] = unjoined_lahman_teams.apply(
+        lambda row: row['name'].split(' ')[-1][:3].upper(),
+        axis=1
+    )
+
+    lahman_columns += ['name_start']
+
+    joined = None
+
+    for join_column in ['manual_teamid', 'teamID', 'franchID', 'teamIDBR', 'initials', 'city_start', 'name_start']:
```

Comment:
some logging would be helpful here, i.e. how many teams matched after trying `manual_teamid`, how many after `teamID`, etc...